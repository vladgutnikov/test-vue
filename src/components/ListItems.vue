<template>
  <v-container>
    <v-row class="justify-center">
      <v-col md="4">
        <v-card max-width="100%" class="mx-auto" v-show="!!items.length">
          <v-list three-line>
            <ListItem
              v-for="(item, index) in items"
              :key="`${index} ${item.header} `"
              :item="item"
            />
          </v-list>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
  import { VCard, VList, VCol, VRow, VContainer } from 'vuetify/lib';
  import ListItem from './ListItem.vue';

  export default {
    name: 'ListItems',
    components: {
      VCard,
      VList,
      VCol,
      VRow,
      VContainer,
      ListItem
    },
    props: {
      items: {
        type: Array,
        default: () => []
      }
    }
  };
</script>
