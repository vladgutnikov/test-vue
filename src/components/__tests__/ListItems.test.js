import { createLocalVue, mount, shallowMount } from '@vue/test-utils';
import ListItems from '../ListItems.vue';
import Vuetify from 'vuetify';
import Vue from 'vue';

describe('ListItems.vue', () => {
  let vuetify;
  const localVue = createLocalVue();

  beforeEach(() => {
    Vue.use(Vuetify);
    vuetify = new Vuetify();
  });

  it('renders a vue instance', () => {
    expect(shallowMount(ListItems)).toBeTruthy();
  });

  it('checking props', () => {
    const newWrapper = mount(ListItems, {
      localVue,
      vuetify,
      propsData: {
        items: [{ title: 'Hello world' }],
      },
    });

    expect(newWrapper.props().items[0]).toEqual({ title: 'Hello world' });
  });

  it('component snapshot ListItems.vue', () => {
    const newWrapper = mount(ListItems, {
      localVue,
      vuetify,
    });

    expect(newWrapper.html()).toMatchSnapshot();
  });
});
