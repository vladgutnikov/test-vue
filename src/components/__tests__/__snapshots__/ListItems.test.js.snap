// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItems.vue should have a custom title and match snapshot 1`] = `
"<div class=\\"container\\">
  <div class=\\"row justify-center\\">
    <div class=\\"col-md-4 col\\">
      <div class=\\"mx-auto v-card v-sheet theme--light\\" style=\\"max-width: 100%; display: none;\\">
        <div role=\\"list\\" class=\\"v-list v-sheet theme--light v-list--three-line\\"></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`ListItems.vue should have match snapshot 1`] = `
"<div class=\\"container\\">
  <div class=\\"row justify-center\\">
    <div class=\\"col-md-4 col\\">
      <div class=\\"mx-auto v-card v-sheet theme--light\\" style=\\"max-width: 100%; display: none;\\">
        <div role=\\"list\\" class=\\"v-list v-sheet theme--light v-list--three-line\\"></div>
      </div>
    </div>
  </div>
</div>"
`;
